app/javascript/components/organisms/StatusHistoryPopover.js

```diff
@@ -0,0 +1,70 @@
+import React, { useState } from 'react'
+import classNames from 'classnames'
+import PropTypes from 'prop-types'
+
+import Icon from '~/components/atoms/Icon'
+import Popover from '~/components/molecules/Popover'
+import StatusItems from '~/components/molecules/lease_up_sidebar/StatusItems'
+import Loading from '../molecules/Loading'
+import apiService from '~/apiService'
+
+const MAX_UPDATES_TO_SHOW_DEFAULT = 4
+const StatusHistoryPopover = ({ applicationId }) => {
+  const [showingAllStatuses, setShowingAllStatuses] = useState(false)
+  const onShowHideStatusesToggled = () => setShowingAllStatuses(!showingAllStatuses)
+
+  const StatusIconButton = ({ onClick, ref }) => (
+    <button ref={ref} onClick={onClick} className='tiny icon margin-left--half'>
+      <Icon icon='list-unordered' size='medium' />
+    </button>
+  )
+  const [data, setData] = useState({ statusItems: [], loading: false })
+
+  const fetchStatusHistory = async () => {
+    try {
```

Comment:
can this be done with promises?

```
setData({ statusItems: data.statusItems, loading: true })
apiService.getFieldUpdateComments(applicationId)
  .then((items) => setData({ statusItems: items, loading: false }))
  .catch(() => setData({ statusItems: data.statusItems, loading: false }))
```

if you still want this to be awaitable you can just return it:
```
// Keeping the async just to label this as a function that returns a promise -- not strictly necessary
const fetchStatusHistory = async () => {
  setData({ statusItems: data.statusItems, loading: true })
  return apiService.getFieldUpdateComments(applicationId)
    .then((items) => setData({ statusItems: items, loading: false }))
    .catch(() => setData({ statusItems: data.statusItems, loading: false }))
}
```